js/classroom.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * A class representing classrooms.
+ * 
+ * @Constructor
+ *
+ * @example
+ *  var classroomExample = {
+ *    classroomCode: "43",
+ *    selected: 1,
+ *    horas_aula: 0,
+ *    vagas_ofertadas: 0,
+ *    vagas_ocupadas: 0,
+ *    alunos_especiais: 0,
+ *    saldo_vagas: 0,
+ *    pedidos_sem_vaga: 0,
+ *    teachers: [
+ *      "First Guy",
+ *      "Second Son",
+ *      "third Weird"
+ *    ],
+ *    schedules: [{@link Schedule}],
+ *    htmlElement: div.classroom-info
+ *  }
+ *
+ * @see UI#createClassroomInfo
+ */
+ // IMPORTANT: the 'ui' variable must be already set up!
+function Classroom(jsonObj, parentLecture) {
+  this.parent = parentLecture;
+  this.teachers = new Array();
+  this.schedules = new Array();
+  if (jsonObj) {
+    this.classroomCode = jsonObj.classroomCode;
+    this.horas_aula = jsonObj.horas_aula;
+    this.vagas_ofertadas = jsonObj.vagas_ofertadas;
+    this.vagas_ocupadas = jsonObj.vagas_ocupadas;
+    this.alunos_especiais = jsonObj.alunos_especiais;
+    this.saldo_vagas = jsonObj.saldo_vagas;
+    this.pedidos_sem_vaga = jsonObj.pedidos_sem_vaga;
+    this.selected = jsonObj.selected;
+    // Array.slice(0) copies the _entire_ array.
```

Comment:
Two comments:

1) This is a shallow copy, is this intended?

2) `array.slice(0)` isn't very legible in my opinion. Maybe create an utils library and add an `Array.prototype.clone()` function there that calls `slice(0)` internally?